src/Language/Fortran/Analysis/SemanticTypes.hs

```diff
@@ -35,10 +35,8 @@ data SemType
   | TCharacter CharacterLen Kind
   | TArray SemType (Maybe Dimensions) -- ^ Nothing denotes dynamic dimensions
```

Comment:
Should this be changed to `TArray SemType Dimensions`? A single dynamic dimension can just be represented by `DimensionsFinalStar` now.

Reply:
Good point. I had thought the `Nothing` option enables declaring an assumed-shape array type (e.g. `integer arr(:)`), but I think I misunderstood them, since the Fortran 2018 spec implies that they tell you the rank/no. of dimensions.

We could change to `TArray SemType (Either Int Dimensions)` to encode assumed-shape arrays. But I'm leaning towards simplifying as you recommend and leaving a comment on assumed-shape arrays. They don't show up until later specs.

Reply:
Ah right, yeah I can't say I'm familiar with any newer way of declaraing assumed size arrays that those standards might allow. If it's only from Fortran 2018 then given fortran-src doesn't support that at all it probably makes sense to leave it for now.

Reply:
It's not from Fortran 2018, that was just the standard I had on hand. Apparently in F90. I'm still shying away from explicitly encoding them for now, since `SemType` tends to deal with F77 code (expanding that leads to extra complexity).

Reply:
Oh hmm, I guess at some point we want to move to something a bit more f90 compliant but that's probably still a while off and can be expanded then (or perhaps sooner if anyone else wants f90 analysis from this).